tests/apps/stateapp/app.go

```diff
@@ -197,6 +234,10 @@ func handler(w http.ResponseWriter, r *http.Request) {
 		res.States = states
 	case "delete":
 		err = deleteAll(req.States)
+	case "upsertTransaction":
+		err = performTransaction(req.States, "upsert")
+	case "deleteTransaction":
+		err = performTransaction(req.States, "delete")
 	default:
```

Comment:
We need to validate upsert + delete combination scenarios.

Reply:
Would you recommend adding an `OperationType string` into the `daprState` struct? This may require a lot of changes to the structs in `stateapp_test.go` when generating test cases.

Reply:
Yes you may need to add operationType to daprState. except for transaction api, you can omit operationType. it doesn't impact many change in the existing tests. If you think it is hard to transaction tests into TestStateApp, you can create new test function separately only for transaction, e.g. TestTransaction(). Please let me know if it doesn't make sense.